File: backend/src/EscalaGcm.Domain/Entities/Ferias.cs
using EscalaGcm.Domain.Common;

namespace EscalaGcm.Domain.Entities;

public class Ferias : AuditableEntity
{
    public int GuardaId { get; set; }
    public DateOnly DataInicio { get; set; }
    public DateOnly DataFim { get; set; }
    public string? Observacao { get; set; }

    public Guarda Guarda { get; set; } = null!;
}

File: frontend/src/api/ferias.ts
import apiClient from './client';
import type { Ferias } from '../types';
export interface CreateFeriasRequest { guardaId: number; dataInicio: string; dataFim: string; observacao?: string; }
export interface UpdateFeriasRequest { guardaId: number; dataInicio: string; dataFim: string; observacao?: string; }
export const getFeriasList = () => apiClient.get<Ferias[]>('/ferias').then(r => r.data);
export const createFerias = (data: CreateFeriasRequest) => apiClient.post<Ferias>('/ferias', data).then(r => r.data);
export const updateFerias = (id: number, data: UpdateFeriasRequest) => apiClient.put<Ferias>(`/ferias/${id}`, data).then(r => r.data);
export const deleteFerias = (id: number) => apiClient.delete(`/ferias/${id}`);
